### PR TITLE
add arabic keyboard

### PR DESCRIPTION
**Pull request by ashrafkamel5**
I have added an Arabic keyboard 
that would be added only when enabling `#define LV_USE_ARABIC_PERSIAN_CHARS 1`
to maintain the library light size

## screenshots
how the Arabic button looks in the other keyboards

image

how the keyboard itself looks

image

## example code
```c
kb = lv_keyboard_create(lv_scr_act(), NULL);
lv_obj_set_event_cb(kb, kb_event_cb);
lv_keyboard_set_mode(kb, LV_KEYBOARD_MODE_TEXT_ARABIC);
```

### CODE REVIEW

File: src/lv_widgets/lv_keyboard.c [hunk 1]
```diff
@@ -37,40 +37,80 @@ static lv_signal_cb_t ancestor_signal;
 static const char * const default_kb_map_lc[] = {"1#", "q", "w", "e", "r", "t", "y", "u", "i", "o", "p", LV_SYMBOL_BACKSPACE, "\n",
                                                  "ABC", "a", "s", "d", "f", "g", "h", "j", "k", "l", LV_SYMBOL_NEW_LINE, "\n",
                                                  "_", "-", "z", "x", "c", "v", "b", "n", "m", ".", ",", ":", "\n",
-                                                 LV_SYMBOL_CLOSE, LV_SYMBOL_LEFT, " ", LV_SYMBOL_RIGHT, LV_SYMBOL_OK, ""
+                                                 LV_SYMBOL_CLOSE, 
+                                                 #if LV_USE_ARABIC_PERSIAN_CHARS == 1
```
Comment (embeddedt) [inline src/lv_widgets/lv_keyboard.c]: Nitpick: please align preprocessor comments to the leftmost column.

File: src/lv_widgets/lv_keyboard.c [hunk 2]
```diff
@@ -354,6 +400,14 @@ void lv_keyboard_def_event_cb(lv_obj_t * kb, lv_event_t event)
         lv_btnmatrix_set_ctrl_map(kb, kb_ctrl[LV_KEYBOARD_MODE_TEXT_LOWER]);
         return;
     }
+    #if LV_USE_ARABIC_PERSIAN_CHARS == 1
+    else if(strcmp(txt, "أب") == 0) {
+        ext->mode = LV_KEYBOARD_MODE_TEXT_UPPER;
```
Comment (embeddedt) [inline src/lv_widgets/lv_keyboard.c]: Why set it to `UPPER` when `LV_KEYBOARD_MODE_TEXT_ARABIC` exists?

File: src/lv_widgets/lv_keyboard.c [hunk 3]
```diff
@@ -87,17 +127,23 @@ static const lv_btnmatrix_ctrl_t default_kb_ctrl_num_map[] = {
...
 /* clang-format on */
 
-static const char * * kb_map[4] = {
+static const char * * kb_map[] = {
     (const char * *)default_kb_map_lc,
     (const char * *)default_kb_map_uc,
     (const char * *)default_kb_map_spec,
     (const char * *)default_kb_map_num
+    #if LV_USE_ARABIC_PERSIAN_CHARS == 1
+    ,(const char * *)default_kb_map_ar
```
Comment (embeddedt) [inline src/lv_widgets/lv_keyboard.c:136]: The comma should be placed after `num`, not on the following line.

**Review by embeddedt — COMMENTED** [review]
Looks good in general. I have one functional comment and a few stylistic ones.

File: src/lv_widgets/lv_keyboard.c (hunk 2 above)
Comment (ashrafkamel5) [reply src/lv_widgets/lv_keyboard.c]: sorry that is a mistake
I don't  know how I didn't notice that

File: src/lv_widgets/lv_keyboard.c (hunk 3 above)
Comment (ashrafkamel5) [reply src/lv_widgets/lv_keyboard.c:136]: why?
the last element in any group isn't followed by a comma 
and the num is the last element
if we didn't use ARABIC_CHARS

File: src/lv_widgets/lv_keyboard.c (hunk 1 above)
Comment (ashrafkamel5) [reply src/lv_widgets/lv_keyboard.c]: ok, on it

Comment (ashrafkamel5) [reply src/lv_widgets/lv_keyboard.c]: I did it but now I think it was more readable as it was first
especially in the arrays

File: src/lv_widgets/lv_keyboard.c (hunk 3 above)
Comment (embeddedt) [reply src/lv_widgets/lv_keyboard.c:136]: The compiler doesn't mind an extra comma at the end of the last element, and it looks neater.

**Comment by kisvegabor** [comment]
Looks good to me, thank you @ashrafkamel5 !

**Comment by kisvegabor** [comment]
I've merged it into the `dev` branch.

**Pull request merged**